src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionIterator.java

```diff
@@ -119,6 +119,8 @@ private AssemblyRegion loadNextAssemblyRegion() {
         while ( locusIterator.hasNext() && nextRegion == null ) {
             final AlignmentContext pileup = locusIterator.next();
 
+            logger.debug(()->"pileup: " + pileup);
```

Comment:
Confirm that `toString()` is lazily evaluated here by doing an empirical test.

Add more context info to the logger messages (eg., Class/method name)

Reply:
Just made the pileup.toString() throw and exception and the integration tests pass... 

Sure though the pileup toString() method has a lot of details in it anyway.